xds/internal/balancer/clusterresolver/resource_resolver.go

```diff
@@ -192,13 +192,15 @@ func (rr *resourceResolver) resolveNow() {
 
 func (rr *resourceResolver) stop() {
 	rr.mu.Lock()
-	defer rr.mu.Unlock()
-	for dm, r := range rr.childrenMap {
-		delete(rr.childrenMap, dm)
-		r.r.stop()
-	}
+	cm := rr.childrenMap
+	rr.childrenMap = make(map[discoveryMechanismKey]resolverMechanismTuple)
```

Comment:
Maybe a comment here as to why we are re-initializing the map here?

Reply:
Done. PTAL?

Reply:
LGTM